block/block.md

```diff
@@ -1 +1,99 @@
+
+* Call `CreateBlock` using executor
+* Sign the block to generate commitment
+* Call `ApplyBlock` using executor to generate an updated state
+* Save the block, validators, and updated state to local store
+* Add the newly generated block to `pendingBlocks` queue
+* Publish the newly generated block to channels to notify other components of the sequencer node (such as block and header gossip)
+
+### Block Publication to DA Network
+
+The block manager of the sequencer full nodes regularly publish the produced blocks (that are pending in the `pendingBlocks` queue) to DA network using the `DABlockTime` configuration parameter defined in the block manager config. In the event of failure to publish the block to DA network, the manager will perform `maxSubmitAttempts` (default is 30) attempts and a exponential backoff interval between the attempts. The exponential backoff interval starts off at 100 milliseconds and it doubles in the next attempt and capped at `DABlockTime`. A successfull publish event leads to emptying of `pendingBlocks` queue and a failure event leads to proper error reporting.
+
+### Block Retrieval from DA Network
+
+The block manager of the full nodes regularly pulls blocks from the DA network at `DABlockTime` intervals and starts off with a DA height read from the last state stored in the local store or `DAStartHeight` configuration parameter, whichever the latest. The block manager also actively maintains and increments the `daHeight` counter after every DA pull. The pull happens by making the `RetrieveBlocks(daHeight)` request using the Data Availability Light Client (DALC) retriever, which can return either `Success`, `NotFound`, or `Error`. In the event of error, a retry logic kicks in after a delay of 100 milliseconds delay between every retries and 10 retries, after which the error is logged and `daHeight` counter is not incremented, which basically results in the intentional stall of the block retrieval logic. In block `NotFound` scenario does not result in an error and it is acceptable to have no rollup block at every DA height. The retrieval successfully increments the `daHeight` counter in this case. Finally, for the `Success` scenario, first the hashes of the blocks which are successfully retrieved are marked as hard confirmed and are sent for application (or state update). The successful state update triggers a fresh DA and blockstore pulls without respecting the `DABlockTime` and `BlockTime` intervals.
+
+#### About Soft/Hard Confirmations
+
+The block manager retrieves blocks from both p2p network and the underlying DA network, because the blocks are available in the p2p network faster and DA retreival is slower (e.g., 1 second vs 15 seconds). The blocks retrieved from the p2p network are only marked as soft confirmed until the DA retrieval succeeds on those blocks at which time they are marked hard confirmed. The hard confirmations can be considered as finality.
```

Comment:
```suggestion
The block manager retrieves blocks from both the p2p network and the underlying DA network because the blocks are available in the p2p network faster and DA retrieval is slower (e.g., 1 second vs 15 seconds). The blocks retrieved from the p2p network are only marked as soft confirmed until the DA retrieval succeeds on those blocks and they are marked hard confirmed. The hard confirmations can be considered to have a higher level of finality.
```